### PR TITLE
IAM-components section

### PR DESCRIPTION
**Pull request by beknazaresenbek**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -401,7 +409,123 @@ via [additional environment variables](https://github.com/FIWARE/helm-charts/blo
...
+
+![IAM-components](./doc/img/iam.png)
+
+* The [Trusted Issuers List](https://github.com/fiware/trusted-issuers-list) service provides an [EBSI Trusted Issuers Registry](https://hub.ebsi.eu/apis/pilot/trusted-issuers-registry/v4) implementation to act as
+the Trusted-List-Service in the DSBA Trust and IAM Framework. In addition, a Trusted-Issuers-List API is provided to
+manage the issuers.
+* [VCVerifier](https://github.com/fiware/vcverifier) provides the necessary endpoints to offer SIOP-2/OIDC4VP compliant authentication flows. It exchanges
+VerifiableCredentials for JWT, that can be used for authorization and authentication in down-stream components.
+* [Credentials Config Service](https://github.com/fiware/credentials-config-service) manages and provides information about services and the credentials they are using. It
+returns the scope to be requested from the wallet per service and the credentials and issuers that are considered to be
```
Comment (pulledtim) [inline README.md:430]: Maybe nitpicking but the CCS doesn't tell you which issuers are trusted, but more like where to check to see if a issuer is trusted/authorized

File: README.md [hunk 2]
```diff
@@ -401,7 +409,123 @@ via [additional environment variables](https://github.com/FIWARE/helm-charts/blo
...
+
+* `rbac` and `serviceAccount`: Depending on your requirements, you might need to adapt settings for RBAC and service
+  account
+* `ingress` or `route`: You need to set up these settings to make a component externally accessible
+* `did`s of participants: Replace/add the DIDs of the issuer and other participants
+* Provide correct key in [keyfile.json](https://github.com/DOME-Marketplace/iam-components/blob/main/charts/iam-components/templates/keycloak.yaml)
+  for your issuer
+* `keycloak.frontendUrl`: Externally accessible address of the keycloak (should be the same as defined in ingress/route)
+* `keycloak.realm`: Adapt clients, users and roles according to your needs
+* `tir.com`: replace everywhere with actual TIR
```
Comment (pulledtim) [inline README.md:478]: Might be confusing as this is a value and the others are field names. But on the other side the intro says "values"... not sure

File: README.md (hunk 1 above)
Comment (beknazaresenbek) [reply README.md:430]: Then the description on the project repo is also wrong: https://github.com/FIWARE/credentials-config-service. Yeah, it's probably better, I'll change it

**Review by pulledtim — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -401,7 +409,123 @@ via [additional environment variables](https://github.com/FIWARE/helm-charts/blo
...
+It returns the scope to be requested from the wallet per service. Furthermore, it specifies the credentials required and
+the issuers list endpoints to validate against, when checking access for a certain service.
+* The [Keycloak-VC-Issuer](https://github.com/fiware/keycloak-vc-issuer) is plugin for Keycloak to support SIOP-2/OIDC4VP clients and issue VerifiableCredentials through
+the OIDC4VCI-Protocol to compliant wallets.
+* [PDP](https://github.com/fiware/dsba-pdp) is an implementation of a Policy-Decision Point, evaluating Json-Web-Tokens containing VerifiableCredentials in a
+DSBA-compliant way. It also supports the evaluation in the context of i4Trust.
+* [Keyrock](https://github.com/ging/fiware-idm) is the FIWARE component responsible for Identity Management. Using Keyrock (in conjunction with other security
+components) enables you to add OAuth2-based authentication and authorization security to your services and applications.
+* [Kong Plugins](https://github.com/fiware/kong-plugins-fiware) allow to extend the API Gateway Kong by further functionalities required for FIWARE-based environments. Kong
```
Comment (dwendland) [inline README.md:438]: I wouldn't state that the plugins only extend Kong for FIWARE-based extensions. 
We should also state here, that there is also the PEP plugin, which is especially required here within the IAM components as PEP component and interacts with the PDP mentioned above.

File: README.md [hunk 4]
```diff
@@ -401,7 +409,123 @@ via [additional environment variables](https://github.com/FIWARE/helm-charts/blo
...
+  order to keep the index manageable.
+
+- install the components using the prepared configuration
+  ```
+    helm install <RELEASE_NAME> dome-iam/iam-components --namespace <NAME_SPACE> --version <CHART_VERSION> -f values.yaml
+  ```
+
+#### How to configure
+
+Before deploying these components, you need to update some values:
```
Comment (dwendland) [inline README.md:468]: I would rephrase this a bit, so that people don't think that these are the only values to adapt. In the end, it is suggested to go through all parameters (the links to the correspondnig charts are provided below to look this up), but these parameters listed below are important to set and should be updated at least. Something like that.
Are we sure the deployment would work, when all other parameters are left with the default values?

File: README.md [hunk 5]
```diff
@@ -401,7 +409,123 @@ via [additional environment variables](https://github.com/FIWARE/helm-charts/blo
...
+VerifiableCredentials for JWT, that can be used for authorization and authentication in down-stream components.
+* [Credentials Config Service](https://github.com/fiware/credentials-config-service) manages and provides information about services and the credentials they are using.
+It returns the scope to be requested from the wallet per service. Furthermore, it specifies the credentials required and
+the issuers list endpoints to validate against, when checking access for a certain service.
+* The [Keycloak-VC-Issuer](https://github.com/fiware/keycloak-vc-issuer) is plugin for Keycloak to support SIOP-2/OIDC4VP clients and issue VerifiableCredentials through
+the OIDC4VCI-Protocol to compliant wallets.
+* [PDP](https://github.com/fiware/dsba-pdp) is an implementation of a Policy-Decision Point, evaluating Json-Web-Tokens containing VerifiableCredentials in a
+DSBA-compliant way. It also supports the evaluation in the context of i4Trust.
+* [Keyrock](https://github.com/ging/fiware-idm) is the FIWARE component responsible for Identity Management. Using Keyrock (in conjunction with other security
```
Comment (dwendland) [inline README.md:436]: Although these sentences are true, we don't use it as IdM component here. Basically what we use Keyrock for is only the iSHARE-compliant Authorization Registry part of it (we might add such link: https://dev.ishare.eu/delegation/endpoint.html), where attribute-based access policies are stored and used during the authorization progress. 
Later on, this will be replaced by an ODRL-compliant policy registry (a description of the policies will follow in the last section of the guide)

File: README.md [hunk 6]
```diff
@@ -401,7 +409,123 @@ via [additional environment variables](https://github.com/FIWARE/helm-charts/blo
...
-> (similar to the Access Node section)
+The [DOME IAM-Framework](https://github.com/DOME-Marketplace/iam-components) is a set of microservices, that enables
+users in the DOME ecosystem to authenticate into the [DOME Marketplace](https://dome-marketplace.org).
+
+#### Overview and subcomponents
+
+The DOME IAM-Framework consists of multiple open-source components. The components are not required to be used, as long
+as alternatives providing the same interfaces are used.
+
+The IAM-Framework consists of following components:
```
Comment (dwendland) [inline README.md:420]: WaltId is not listed below. Although we won't need it soon anymore, at the moment it is required. So we should also put a link here for consistency.

File: README.md [hunk 7]
```diff
@@ -401,7 +409,123 @@ via [additional environment variables](https://github.com/FIWARE/helm-charts/blo
 
 ### IAM components
 
-> (similar to the Access Node section)
+The [DOME IAM-Framework](https://github.com/DOME-Marketplace/iam-components) is a set of microservices, that enables
+users in the DOME ecosystem to authenticate into the [DOME Marketplace](https://dome-marketplace.org).
```
Comment (dwendland) [inline README.md:413]: Maybe we could add a sentence, that the authentication process itself is further described below in the #authentication section

File: README.md [hunk 8]
```diff
@@ -401,7 +409,123 @@ via [additional environment variables](https://github.com/FIWARE/helm-charts/blo
...
+
+#### How to configure
+
+Before deploying these components, you need to update some values:
+
+* `rbac` and `serviceAccount`: Depending on your requirements, you might need to adapt settings for RBAC and service
+  account
+* `ingress` or `route`: You need to set up these settings to make a component externally accessible
+* `did`s of participants: Replace/add the DIDs of the issuer and other participants
+* Provide correct key in [keyfile.json](https://github.com/DOME-Marketplace/iam-components/blob/main/charts/iam-components/templates/keycloak.yaml)
```
[the remark this answers is not in the document]
Comment (beknazaresenbek) [reply README.md]: No, only for did:key, so I've updated the description here

File: README.md (hunk 4 above)
Comment (beknazaresenbek) [reply README.md]: Not really, the accessible URLs still need to be replaced, and I didn't set any rbac and serviceAccount

**Review by dwendland — APPROVED** [review]
